### ISSUE TITLE
Add fields to World to contain robot status and change SensorFusion::update_world function accordingly

### ISSUE THREAD

**Issue by jonathanlew** [UBC-Thunderbots/Software#1149]
### Description of the task
Currently, although sensor fusion has access to the robot status, it doesn't communicate that to AI. This should be communicated through a more complete World object. 

### Acceptance criteria

- [ ] Add fields to World to contain robot status and change SensorFusion::update_world function accordingly
- [ ] testing

**Comment by MathewMacDougall** [comment]
This only really affects the Robot class, and has implications such as differentiating friendly from enemy robots (ie. is doesn't make sense for them to be the same class since now they will contain much different info), so we should probably have a meeting about this beforehand.

We may also want to better define what `RobotStatus` should include so this also may be partially blocked by moving to proto, depending what our RobotStatus abstraction is.

**Comment by jonathanlew** [comment]
Superseded by #1164
